File: src/sim/measurements_options.cc
/* src/sim/intervalsize_l3.cc : 
This object is used to pass value to global variable intervalsize used in src/sim/mypriority.cc. 
 */

#include "sim/measurements_options.hh"

#include<iostream>

#include "base/trace.hh"
#include "base/statistics.hh"

namespace gem5
{

MeasurementsOptions::MeasurementsOptions(const MeasurementsOptionsParams &p) : 
	SimObject(p),
	num_cores(p.num_cores),
	num_llcs(p.num_llcs),
	llc_dump_interval(p.llc_dump_interval)
{
	measurements::setMeasurementsOptions((int)num_cores, (int)num_llcs,(int)llc_dump_interval);
}
}//namespace gem5

File: src/sim/MeasurementsOptions.py
##This object is used to pass value to global variable intervalsize used in src/sim/mesurements.cc. 


from m5.SimObject import SimObject 
from m5.params import *

class MeasurementsOptions(SimObject):
    type = 'MeasurementsOptions'
    cxx_header = 'sim/measurements_options.hh'
    cxx_class="gem5::MeasurementsOptions"

    num_cores = Param.Int('8', "Number of cores being recorded");
    num_llcs = Param.Int('2', "Number of LLCs to record");
    llc_dump_interval = Param.Int('1000000', "LLC dump interval");
